src/osd/ReplicatedBackend.cc

```diff
@@ -1177,16 +1177,18 @@ void ReplicatedBackend::repop_applied(RepModifyRef rm)
 
   const MOSDRepOp *req = static_cast<const MOSDRepOp*>(m);
   version = req->version;
-  if (!rm->committed)
-    ack = new MOSDRepOpReply(
-	static_cast<const MOSDRepOp*>(m), parent->whoami_shard(),
-	0, get_osdmap()->get_epoch(), CEPH_OSD_FLAG_ACK);
-
-  // send ack to acker only if we haven't sent a commit already
-  if (ack) {
-    ack->set_priority(CEPH_MSG_PRIO_HIGH); // this better match commit priority!
-    get_parent()->send_message_osd_cluster(
-      rm->ackerosd, ack, get_osdmap()->get_epoch());
+  if (!store->is_synchronous_apply()) {
+    if (!rm->committed)
+      ack = new MOSDRepOpReply(
+	  static_cast<const MOSDRepOp*>(m), parent->whoami_shard(),
+	  0, get_osdmap()->get_epoch(), CEPH_OSD_FLAG_ACK);
+
+    // send ack to acker only if we haven't sent a commit already
+    if (ack) {
+      ack->set_priority(CEPH_MSG_PRIO_HIGH); // this better match commit priority!
+      get_parent()->send_message_osd_cluster(
+	  rm->ackerosd, ack, get_osdmap()->get_epoch());
+    }
```

Comment:
I would take this opportunity to combine the who if statements (the if (ack) is redundant), and just do a single if (!rm->committed && !store->is_synchronous_apply()) { ... }.  Otherwise, this looks right!